### PR TITLE
Add `with-xata` example

### PR DESCRIPTION
**Pull request by atilafassina**
This Pull Request adds a `with-xata` example to use [Xata](https://xata.io) as the data layer of a Next.js app.

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/with-xata/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+<h1> Next.js + Xata Minimal Template <img alt="Xatafly, the logo from Xata" src="/public/flap.gif" width="50" /></h1>
```
Comment (leerob) [inline examples/with-xata/README.md:1]: Can we use the same README format as the other examples, please? 🙏

File: examples/with-xata/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "main": "index.js",
```
Comment (leerob) [inline examples/with-xata/package.json:2]:
```suggestion
```

File: examples/with-xata/pages/_app.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import type { AppProps } from 'next/app'
+import Head from 'next/head'
+import '../styles/root.css'
+
+const metas = {
+  title: 'Next.js with-xata',
+  description: 'Run Next.js with Xata with this awesome template',
+  image:
+    process.env.NODE_ENV === 'development'
```
Comment (leerob) [inline examples/with-xata/pages/_app.tsx:9]: Why is this needed?

**Review by leerob — COMMENTED** [review]
Looking really good!

File: examples/with-xata/pages/_app.tsx (hunk 3 above)
Comment (atilafassina) [reply examples/with-xata/pages/_app.tsx:9]: because OG:Image requires it to be absolute urls, pointing it to localhost on dev makes it easier to test a new image, I thought it was a nice addition to whoever bases on this template to build something.

But absolutely not _needed_

File: examples/with-xata/README.md (hunk 1 above)
Comment (atilafassina) [reply examples/with-xata/README.md]: I just pushed some changes to the Readme, I think now it's more aligned with the rest of the examples

**Comment by atilafassina** [comment]
Heeeey, folks!

Gentle little poke on this. Anything I can do to help it get merged?? 🙏

File: examples/with-xata/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Next.js + Xata
+
+This example showcases how to use Next.js with [Xata](https://xata.io) as your data layer.
```
Comment (leerob) [inline examples/with-xata/README.md:3]: The README still needs the follow the same format as other examples please.

Comment (leerob) [reply examples/with-xata/README.md:3]: Can merge after that!

Comment (balazsorban44) [reply examples/with-xata/README.md:3]: Addressed!

**Review by balazsorban44 — APPROVED** [review]
